### ISSUE TITLE
CloudWatch: Switch to LogGroupIdentifier in GetLogGroupFields request

### ISSUE THREAD

**Issue by sunker** [grafana/grafana#61257]
While refactoring the log group fields request in [this PR](https://github.com/grafana/grafana/pull/60909), I noticed this request is not yet cross-account compatible. In order to support cross account querying, it needs to use [LogGroupIdentifier instead of LogGroupName](https://github.com/grafana/grafana/blob/c72ab2109613510ae266fce7e25ab8d5f212d401/pkg/tsdb/cloudwatch/services/log_groups.go#L59-L63) as input in the call to the GetLogGroupFields api. 

While working with the refactoring, I wasn't able to use LogGroupIdentifier since there was a bug in the API that did not allow **only** specifying this field - you also needed to specify LogGroupName. AWS have [confirmed](https://raintank-corp.slack.com/archives/C029UNXDFQV/p1673280482649849) this is a bug on their side, I they plan to fix it soon. Once it's fixed, we should switch to use LogGroupIdentifier if it's defined in the LogGroupFieldsRequest. When doing so, one can also uncomment [this](https://github.com/grafana/grafana/blob/main/pkg/tsdb/cloudwatch/services/log_groups_test.go#L247-L265) unit test.

**Comment by sladyn98** [comment]
```
type LogGroupFieldsRequest struct {
	ResourceRequest
	LogGroupName string
	LogGroupARN  string
}
```

If aws has fixed this we could refactor the code using LogGroupIdentifier

**Comment by sarahzinger** [comment]
During a recent backend refinement session, I said I'd take a look to learn more about what this is. Basically we have a bit of code clean up to do to let us take advantage of the new cross account observability when using the code editor. This was originally blocked by an issue on AWS's end which I believe has been fixed since this ticket was made. Seems like a nice to have.
